### ISSUE TITLE
Clipboard API mixies representations of an item with distinct items.

### ISSUE THREAD

**Issue by rniwa** [w3c/clipboard-apis#55]
On macOS and iOS, there could be multiple representations of multiple entires for copy & paste and drag & drop.

For example, [UIPasteboard](https://developer.apple.com/documentation/uikit/uipasteboard?language=objc) exposes [items](https://developer.apple.com/documentation/uikit/uipasteboard/1622067-items?language=objc) property on iOS, and each item represents multiple representations (e.g. PNG, JPEG, etc...) of a single entity (the same image).

The current [DataTransferItemList](https://html.spec.whatwg.org/multipage/dnd.html#datatransferitemlist) representation doesn't sit well with this model because it conflates different representations of a single entry with multiple entities in the clipboard.

**Comment by alecazam** [comment]
There is no way to specify that an item has multiple flavors/representations on the clipboard. mozGetDataAt and mozSetDataAt got this correct with an index that was in common across the flavors of a given item. This is also reflective of the native clipboard API.  Adding a series of flavors to the clipboard as individual items of a DataTransfer object doesn't help the receiver determine if they're a flavor or unique item.  Electron has a similar problem with only support for single item, multiple flavors.

Examples:
items: foo.png foo.jpg <- single item, two flavors
items: foo.png bar.jpg <- multiple items, two unique items
items: foo.png foo.jpg, bar.png bar.jpg <- multiple items, two flavors of the each item

In the DataTransfer item list, and index is needed to collect items.
item[0]             item[1]
index: 0            index: 0  <- this is what is missing
name: foo.png name: foo.jpg  <- flavors of same content
blob/file           blob/file

**Comment by rniwa** [comment]
Right. We need a better API which differentiates the flavor/representation of an item in the native clipboard from distinct entires/items in them.

**Comment by garykac** [comment]
Proposal: https://docs.google.com/document/d/1sMcajoYAQ_M_VOIn1RLokG1sVB7skjoeviqt9aZOcDk/edit
